### PR TITLE
releases: add 4.0.0-rc release notes

### PR DESCRIPTION
**Pull request by toutdesuite**
### What is changed, added or deleted? (Required)

add 4.0.0-rc release notes

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/2653
- Other reference link(s):

### CODE REVIEW

**Comment by scsldb** [comment]
LGTM

File: releases/4.0.0-rc.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+# TiDB 4.0 RC Release Notes
+
+Release date: April 8, 2020
+
+TiDB version: 4.0.0-rc
+
+TiDB Ansible version: 4.0.0-rc
+
+## TiDB 4.0.0-rc Compatibility Changes
```
Comment (lilin90) [inline releases/4.0.0-rc.md:14]: Sounds redundant to me here.
```suggestion
## Compatibility Changes
```

File: releases/4.0.0-rc.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    - Support the `pipelined` feature in pessimistic transactions, which improves the TPC-C performance by 20%. The risk is that the transaction commit might fail because of lock failure during the execution [#6984](https://github.com/tikv/tikv/pull/6984)
+    - Enable the `unify-read-pool` configuration item in new clusters by default and use the previous setting of this item in old clusters [#7059](https://github.com/tikv/tikv/pull/7059)
+
++ Tools
+
+    - TiDB Binlog
+
+        * Add the configuration item for verifying Common Name [#934](https://github.com/pingcap/tidb-binlog/pull/934)
+
+## TiDB 4.0.0-rc Important Bug Fixes
```
Comment (lilin90) [inline releases/4.0.0-rc.md:31]:
```suggestion
## Important Bug Fixes
```

File: releases/4.0.0-rc.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        * Fix the issue that an error is returned if the `test` schema does not exist in the downstream when MySQL sink is executing the DDL statement [#353](https://github.com/pingcap/ticdc/pull/353)
+        * Support the real-time interactive mode in CDC cli [#351](https://github.com/pingcap/ticdc/pull/351)
+        * Support checking whether the table in the upstream can be replicated during data replication [#368](https://github.com/pingcap/ticdc/pull/368)
+        * Support asynchronous write to Kafka [#344](https://github.com/pingcap/ticdc/pull/344)
+
++ TiDB Ansible
+
+    - Optimize the prompts by indicating users to upgrade the Linux kernel version to resolve the CPU epollexclusive check failure, with the earliest supported version [#1230](https://github.com/pingcap/tidb-ansible/pull/1230)
+    - Fix the issue of failed upgrade to tidb-ansible 4.0 by optimizing the Dashboard initialization [#1237](https://github.com/pingcap/tidb-ansible/pull/1237)
+    - Optimize the creation of the TiFlash directory [#1239](https://github.com/pingcap/tidb-ansible/pull/1239)
```
Comment (lilin90) [inline releases/4.0.0-rc.md:106]: Remove?

File: releases/4.0.0-rc.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+category: Releases
+---
+
+# TiDB 4.0 RC Release Notes
+
+Release date: April 8, 2020
+
+TiDB version: 4.0.0-rc
+
+TiDB Ansible version: 4.0.0-rc
```
Comment (lilin90) [inline releases/4.0.0-rc.md:12]: Need update to TiUP info.

**Comment by TomShawn** [comment]
@scsldb @lilin90 Comments addressed, PTAL again, thanks!

**Pull request merged**